### DISCUSSION TITLE
How to use queryInterface.insert without a Model instance.

### DISCUSSION THREAD

**Discussion by IDrumsey** [sequelize/sequelize#17088] [Help & Questions] [answered]
Is there a way to use the `queryInterface` without passing a Model instance?

I'm currently using this code in a seeder

```javascript
await queryInterface.insert(null, "MyTable", {
    myCol1: 123
})
```

and I'm getting the error

```bash
ERROR: Validation failed for parameter '0'. Invalid string.
```

The ts signature shows you can pass null

```typescript
public insert(instance: Model | null, tableName: string, values: object, options?: QueryOptions): Promise<object>;
```

**Comment by ephys** [comment, marked as answer]
`queryInterface.insert` is not really designed to be used directly. We've been working on making all `queryInterface` methods publicly usable in Sequelize 7. This is the PR related to reworking insert: https://github.com/sequelize/sequelize/pull/16988

**Reply by IDrumsey** [reply]
`queryInterface` is all over the migrations docs though - https://sequelize.org/docs/v6/other-topics/migrations/.

**Reply by ephys** [reply]
Yup! Which is why it's a bit crazy that it's full of methods designed for internal use only (not that I'm blaming the past maintainers for their design decisions. Sequelize wouldn't be where it is without them), and why we've been working on fixing that in v7

**Reply by IDrumsey** [reply]
Oh ok, so some methods in `queryInterface` should be used and some should not be used directly?
